lib/internal/http2/util.js

```diff
@@ -178,6 +180,119 @@ const IDX_OPTIONS_MAX_OUTSTANDING_SETTINGS = 7;
+const IDX_STREAM_STATS_SENTBYTES = 4;
+const IDX_STREAM_STATS_RECEIVEDBYTES = 5;
+
+const IDX_SESSION_STATS_TYPE = 0;
+const IDX_SESSION_STATS_PINGRTT = 1;
+const IDX_SESSION_STATS_FRAMESRECEIVED = 2;
+const IDX_SESSION_STATS_FRAMESSENT = 3;
+const IDX_SESSION_STATS_STREAMCOUNT = 4;
+const IDX_SESSION_STATS_STREAMAVERAGEDURATION = 5;
+const IDX_SESSION_STATS_DATA_SENT = 6;
+const IDX_SESSION_STATS_DATA_RECEIVED = 7;
+const IDX_SESSION_STATS_MAX_CONCURRENT_STREAMS = 8;
+
+function collectStats(entry) {
+  switch (entry.name) {
+    case 'Http2Stream':
+      collectStreamStats(entry);
+      break;
+    case 'Http2Session':
+      collectSessionStats(entry);
+      break;
+  }
+}
+
+function collectStreamStats(entry) {
+  Object.defineProperties(entry, {
+    id: {
+      configurable: true,
+      enumerable: true,
+      value: streamStats[IDX_STREAM_STATS_ID] >>> 0
+    },
+    timeToFirstByte: {
+      configurable: true,
+      enumerable: true,
+      value: streamStats[IDX_STREAM_STATS_TIMETOFIRSTBYTE]
+    },
+    timeToFirstHeader: {
+      configurable: true,
+      enumerable: true,
+      value: streamStats[IDX_STREAM_STATS_TIMETOFIRSTHEADER]
+    },
+    timeToFirstByteSent: {
+      configurable: true,
+      enumerable: true,
+      value: streamStats[IDX_STREAM_STATS_TIMETOFIRSTBYTESENT]
+    },
+    sentBytes: {
+      configurable: true,
+      enumerable: true,
+      value: streamStats[IDX_STREAM_STATS_SENTBYTES]
+    },
+    receivedBytes: {
+      configurable: true,
+      enumerable: true,
+      value: streamStats[IDX_STREAM_STATS_RECEIVEDBYTES]
+    }
+  });
+}
+
+function collectSessionStats(entry) {
+  Object.defineProperties(entry, {
+    type: {
+      configurable: true,
+      enumerable: true,
+      value: sessionName(sessionStats[IDX_SESSION_STATS_TYPE] >>> 0)
+    },
+    pingRTT: {
+      configurable: true,
+      enumerable: true,
+      value: sessionStats[IDX_SESSION_STATS_PINGRTT]
+    },
+    framesReceived: {
+      configurable: true,
+      enumerable: true,
+      value: sessionStats[IDX_SESSION_STATS_FRAMESRECEIVED]
+    },
+    framesSent: {
+      configurable: true,
+      enumerable: true,
+      value: sessionStats[IDX_SESSION_STATS_FRAMESSENT]
+    },
+    streamCount: {
+      configurable: true,
+      enumerable: true,
+      value: sessionStats[IDX_SESSION_STATS_STREAMCOUNT]
+    },
+    streamAverageDuration: {
+      configurable: true,
+      enumerable: true,
+      value: sessionStats[IDX_SESSION_STATS_STREAMAVERAGEDURATION]
+    },
+    sentBytes: {
+      configurable: true,
+      enumerable: true,
+      value: sessionStats[IDX_SESSION_STATS_DATA_SENT]
+    },
+    receivedBytes: {
+      configurable: true,
+      enumerable: true,
+      value: sessionStats[IDX_SESSION_STATS_DATA_RECEIVED]
+    },
+    maxConcurrentStreams: {
+      configurable: true,
+      enumerable: true,
+      value: sessionStats[IDX_SESSION_STATS_MAX_CONCURRENT_STREAMS]
+    }
+  });
```

Comment:
Just curious, why not use plain writable properties? Is that what we do in the rest of the `perf` API?